### ISSUE TITLE
Add `processed-by` entry to the producer

### ISSUE THREAD

**Issue by zetanumbers** [bytecodealliance/wizer#105]
This is probably useful to add.

https://docs.rs/wasm-metadata/0.206.0/wasm_metadata/struct.Producers.html
https://github.com/WebAssembly/tool-conventions/blob/main/ProducersSection.md

**Comment by fitzgen** [comment]
Yeah, this would be good to do. https://docs.rs/wasm-encoder/latest/wasm_encoder/struct.ProducersSection.html is the relevant API for the encoding library we use, if you want to try your hand at it.
